In [3]:
import pandas as pd
import numpy as np
fields=['appln_id', 'cpc_class_symbol']
#Extreme Temperatures
df = pd.read_csv('/Volumes/NJ_4TB/PATSAT/MyFiles/Extras/Adapt.csv', skipinitialspace=True,chunksize=1000000, usecols = fields, header=0)
header = True
for chunk in df:
    df1 = chunk[chunk['cpc_class_symbol'].str.contains('Y02A30/24|Y02A30/254|Y02A30/27|Y02A30/3|Y02A30/6|Y02A40/13|Y02A40/146|Y02A40/25|Y02A50/3|Y02A40/963')]
    df1.to_csv('/Volumes/NJ_4TB/PATSAT/MyFiles/Extras/1_ExTemp.csv', index=False, mode='a', header=header)
    header = False
    del df1

In [5]:
#Storms
df = pd.read_csv('/Volumes/NJ_4TB/PATSAT/MyFiles/Extras/Adapt.csv', skipinitialspace=True,chunksize=1000000, usecols = fields, header=0)
header = True
for chunk in df:
    df3 = chunk[chunk['cpc_class_symbol'].str.contains('Y02A10/3|Y02A10/4|Y02A30/14|Y02A30/3|Y02A30/6')]
    df3.to_csv('/Volumes/NJ_4TB/PATSAT/MyFiles/Extras/3_Storm.csv', index=False, mode='a', header=header)
    header = False
    del df3

In [6]:
#Floods
df = pd.read_csv('/Volumes/NJ_4TB/PATSAT/MyFiles/Extras/Adapt.csv', skipinitialspace=True,chunksize=1000000, usecols = fields, header=0)
header = True
for chunk in df:
    df4 = chunk[chunk['cpc_class_symbol'].str.contains('Y02A10/11|Y02A10/3|Y02A10/4|Y02A30/14|Y02A30/3|Y02A30/6')]
    df4.to_csv('/Volumes/NJ_4TB/PATSAT/MyFiles/Extras/4_Flood.csv', index=False, mode='a', header=header)
    header = False
    del df4

In [7]:
#Landslide
df = pd.read_csv('/Volumes/NJ_4TB/PATSAT/MyFiles/Extras/Adapt.csv', skipinitialspace=True,chunksize=1000000, usecols = fields, header=0)
header = True
for chunk in df:
    df5 = chunk[chunk['cpc_class_symbol'].str.contains('Y02A10/23|Y02A30/3|Y02A30/6')]
    df5.to_csv('/Volumes/NJ_4TB/PATSAT/MyFiles/Extras/5_Landslide.csv', index=False, mode='a', header=header)
    header = False
    del df5

In [8]:
#WaveAction
df = pd.read_csv('/Volumes/NJ_4TB/PATSAT/MyFiles/Extras/Adapt.csv', skipinitialspace=True,chunksize=1000000, usecols = fields, header=0)
header = True
for chunk in df:
    df6 = chunk[chunk['cpc_class_symbol'].str.contains('Y02A10/26|Y02A30/3|Y02A30/6')]
    df6.to_csv('/Volumes/NJ_4TB/PATSAT/MyFiles/Extras/6_WaveAct.csv', index=False, mode='a', header=header)
    header = False
    del df6

In [9]:
#Drought
df = pd.read_csv('/Volumes/NJ_4TB/PATSAT/MyFiles/Extras/Adapt.csv', skipinitialspace=True,chunksize=1000000, usecols = fields, header=0)
header = True
for chunk in df:
    df7 = chunk[chunk['cpc_class_symbol'].str.contains('Y02A20/|Y02A40/13|Y02A40/146|Y02A40/22')]
    df7.to_csv('/Volumes/NJ_4TB/PATSAT/MyFiles/Extras/7_Drought.csv', index=False, mode='a', header=header)
    header = False
    del df7

In [10]:
#Wildfire
df = pd.read_csv('/Volumes/NJ_4TB/PATSAT/MyFiles/Extras/Adapt.csv', skipinitialspace=True,chunksize=1000000, usecols = fields, header=0)
header = True
for chunk in df:
    df9 = chunk[chunk['cpc_class_symbol'].str.contains('Y02A30/14|Y02A30/3|Y02A30/6')]
    df9.to_csv('/Volumes/NJ_4TB/PATSAT/MyFiles/Extras/9_Wildfire.csv', index=False, mode='a', header=header)
    header = False
    del df9

In [3]:
#merge with larger file
import pandas as pd
import numpy as np
Disaster = ['1_ExTemp', '3_Storm', '4_Flood', '5_Landslide', '6_WaveAct', '7_Drought', '9_Wildfire']
for d in Disaster:
    df = pd.read_csv('/Volumes/NJ_4TB/PATSAT/MyFiles/Adapt_everything.csv', skipinitialspace=True,chunksize=1000000, header=0)
    df1 = pd.read_csv('/Volumes/NJ_4TB/PATSAT/MyFiles/' + d +'.csv', skipinitialspace=True,header=0)
    for chunk in df:
        df2 = pd.merge(df1, chunk, on="appln_id", how = 'inner')
        df2.to_csv('/Volumes/NJ_4TB/PATSAT/MyFiles/Extras/' + d + '_everything.csv', encoding='utf-8', index=False)
        del df1, df2

In [ ]:
df1 = pd.read_csv('/Volumes/NJ_4TB/PATSAT/MyFiles/5_Landslide.csv', skipinitialspace=True,header=0)
df1.shape[0]

In [1]:
import pandas as pd
import numpy as np
Disaster = ['1_ExTemp', '3_Storm', '4_Flood', '5_Landslide', '6_WaveAct', '7_Drought', '9_Wildfire']
for d in Disaster:
    df1 = pd.read_csv('/Volumes/NJ_4TB/PATSAT/MyFiles/Extras/' + d + '_everything.csv', skipinitialspace=True,header=0)
    n = df1['appln_id'].nunique()
    print("There are "+ str(n) + " unique patents to counter " + str(d))
    del df1, n

There are 84737 unique patents to counter 1_ExTemp
There are 19822 unique patents to counter 3_Storm
There are 19822 unique patents to counter 4_Flood
There are 11308 unique patents to counter 5_Landslide
There are 11362 unique patents to counter 6_WaveAct
There are 56888 unique patents to counter 7_Drought
There are 16629 unique patents to counter 9_Wildfire


There are 84737 unique patents to counter 1_ExTemp

There are 19822 unique patents to counter 3_Storm

There are 19822 unique patents to counter 4_Flood

There are 11308 unique patents to counter 5_Landslide

There are 11362 unique patents to counter 6_WaveAct

There are 56888 unique patents to counter 7_Drought

There are 16629 unique patents to counter 9_Wildfire
